src/typechecker/FStar.TypeChecker.Normalize.fs

```diff
@@ -81,8 +95,7 @@ type stack_elt =
  | CBVApp   of env * term * aqual * Range.range
  | Meta     of env * S.metadata * Range.range
  | Let      of env * binders * letbinding * Range.range
- | Cfg      of cfg
- | Debug    of term * BU.time
+ | Cfg      of cfg * option<(term * BU.time)>
```

Comment:
This change fuses the Debug and Cfg nodes, by adding an optional debug value in a Cfg. 

The motivation for this is that prior to this change, if you turned on debugging using --debug_level print_normalized_terms, then you would actually change the behavior of the normalizer ... which is crazy.

The reason for this behavior was that adding a Debug marker on the stack causes many invariants to be broken. Fusing the debug marker with the Cfg marker ensures that debugging is handled wherever Cfg changes take place, allowing debug info to be emitted without changing the behavior.

Moral of the story: Don't add another kind of element on the stack, unless you're really sure that you are really consuming that element off the stack everywhere you need to.